src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -1217,4 +1212,37 @@ namespace winrt::TerminalApp::implementation
             itemContainer.DataContext(args.Item());
         }
     }
+
+    // Method Description:
+    // - Reads the list of recent commands from the persistent application state
+    // Return Value:
+    // - The list of FilteredCommand representing the ones stored in the state
+    Windows::Foundation::Collections::IVector<winrt::TerminalApp::FilteredCommand> CommandPalette::_loadRecentCommands() const
+    {
+        const auto parsedCommands = winrt::single_threaded_vector<winrt::TerminalApp::FilteredCommand>();
+        const auto recentCommands = ApplicationState::SharedInstance().RecentCommands();
+        std::for_each(begin(recentCommands), end(recentCommands), [&](const auto& c) {
+            auto parsedCommand = _buildCommandLineCommand(c.c_str());
+            if (parsedCommand.has_value())
+            {
+                parsedCommands.Append(parsedCommand.value());
+            }
+        });
+        return parsedCommands;
```

Comment:
```suggestion
        const auto recentCommands = ApplicationState::SharedInstance().RecentCommands();

        std::vector<winrt::TerminalApp::FilteredCommand> parsedCommands;
        parsedCommands.reserve(recentCommands.Size());

        for (const auto& c : recentCommands) {
            if (auto parsedCommand = _buildCommandLineCommand(c))
            {
                parsedCommands.Append(*parsedCommand);
            }
        }

        return winrt::single_threaded_vector(std::move(parsedCommands));
```

This does a few things:
* reserve the required memory upfront for `parsedCommands`
* STL classes are significantly faster than COM classes (~3x)
* for-range loop, which compilers _love_ and is easier to read than `std::for_each` (IMO)
* if with init-statement, means we don't have to ask for `has_value()` again, because the if condition will check for truthy-ness by default

Reply:
I'm curious, is a `FilteredCommand` something that is easy to serialize/deserialize? If so, this could all just be
```
return Application::SharedInstance().RecentCommands();
```

Reply:
Love it. Agree with every point! Will do it!